ReactWindows/ReactNative.Shared/UIManager/ThemedReactContext.cs

```diff
@@ -1,75 +1,114 @@
 
-        /// <summary>
-        /// Gets the instance of the <see cref="IJavaScriptModule"/> associated
-        /// with the <see cref="IReactInstance"/>.
-        /// </summary>
-        /// <typeparam name="T">Type of JavaScript module.</typeparam>
-        /// <returns>The JavaScript module instance.</returns>
+        /// <inheritdoc />
         public T GetJavaScriptModule<T>()
             where T : IJavaScriptModule, new()
         {
             return _reactContext.GetJavaScriptModule<T>();
         }
 
-        /// <summary>
-        /// Adds a lifecycle event listener to the context.
-        /// </summary>
-        /// <param name="listener">The listener.</param>
+        /// <inheritdoc />
         public void AddLifecycleEventListener(ILifecycleEventListener listener)
         {
             _reactContext.AddLifecycleEventListener(listener);
         }
 
-        /// <summary>
-        /// Removes a lifecycle event listener from the context.
-        /// </summary>
-        /// <param name="listener">The listener.</param>
+        /// <inheritdoc />
         public void RemoveLifecycleEventListener(ILifecycleEventListener listener)
         {
             _reactContext.RemoveLifecycleEventListener(listener);
         }
+
+        /// <inheritdoc />
+        public void AddBackgroundEventListener(IBackgroundEventListener listener)
+        {
+            _reactContext.AddBackgroundEventListener(listener);
+        }
+
+        /// <inheritdoc />
+        public void RemoveBackgroundEventListener(IBackgroundEventListener listener)
+        {
+            _reactContext.RemoveBackgroundEventListener(listener);
+        }
+
+        /// <inheritdoc />
+        public bool IsOnDispatcherQueueThread()
+        {
+            return _reactContext.IsOnDispatcherQueueThread();
+        }
+
+        /// <inheritdoc />
+        public void RunOnDispatcherQueueThread(Action action)
+        {
+            _reactContext.RunOnDispatcherQueueThread(action);
+        }
+
+        /// <inheritdoc />
+        public bool IsOnJavaScriptQueueThread()
+        {
+            return _reactContext.IsOnJavaScriptQueueThread();
+        }
+
+        /// <inheritdoc />
+        public void RunOnJavaScriptQueueThread(Action action)
+        {
+            _reactContext.RunOnJavaScriptQueueThread(action);
+        }
+
+        /// <inheritdoc />
+        public bool IsOnNativeModulesQueueThread()
+        {
+            return _reactContext.IsOnNativeModulesQueueThread();
+        }
+
+        /// <inheritdoc />
+        public void RunOnNativeModulesQueueThread(Action action)
+        {
+            _reactContext.RunOnNativeModulesQueueThread(action);
+        }
+
+        /// <inheritdoc />
+        public void HandleException(Exception exception)
+        {
+            _reactContext.HandleException(exception);
+        }
+
+        public Task DisposeAsync()
+        {
+            return (_reactContext as IAsyncDisposable)?.DisposeAsync();
+        }
```

Comment:
We don't need to expose the `DisposeAsync`

Reply:
are you aware of a way to make it non-public when it's inherited from an interface? I already pushed the interface inheritance down from the IReactContext to the concrete children as part of the previous feedback. I'm not sure what to do here, and I'm pretty sure the original implementation will be necessary later based on my private branch. Let me know how exactly to proceed.